### PR TITLE
WIP AArch64: Implementation of arraychk evaluator

### PR DESCRIPTION
**Pull request by marufunb**
This commit implements arrayCHK evaluator for openj9.

### CODE REVIEW

**Comment by marufunb** [comment]
@0xdaryl I have implemented the arraychkevaluator. Can you please manage sometime to have a look in the implementation?

File: runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -114,6 +162,189 @@ J9::ARM64::TreeEvaluator::DIVCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
+   // Same array, we are done.
+   //
+   generateTrg1Src2Instruction(cg,TR::InstOpCode::ccmpw, node, cndReg, obj1Reg, obj2Reg);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, doneLabel, TR::CC_EQ, cndReg);
+
+   // If we know nothing about either object, test object1 first. It has to be an array.
+   //
+   if (!node->isArrayChkPrimitiveArray1() && !node->isArrayChkReferenceArray1() && !node->isArrayChkPrimitiveArray2() && !node->isArrayChkReferenceArray2())
+      {
+#ifdef OMR_GC_COMPRESSED_POINTERS
```
Comment (aarongraham9) [inline runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: Do we care about compressed pointers at this point? Is this an optimization?

Comment (marufunb) [reply runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp]: I need Daryl's feedback in this. Initially I have tried to complete the implementation and still waiting for Daryl's feedback on this.

**Comment by marufunb** [comment]
This PR (#3858) has been superseded by #4214. Closing.

**Comment by marufunb** [comment]
This PR (#6297) has been superseded by #6786. Closing.

**Pull request closed without merging**